### ISSUE TITLE
vpcc self-update silently skips Python-code updates; doctor doesn't warn about pipx package version skew

### ISSUE THREAD

**Issue by ademczuk** [VoidChecksum/void-patcher-cc#19]
## Summary

`vpcc self-update` only syncs `patches/*.json` (per its help text). The vpcc Python package itself (vpcc/, install.ps1, contrib/) does not update. Users running `vpcc self-update` see "already up to date" when the patch SHA matches, even if their pipx-installed vpcc Python code is several releases behind remote. Doctor doesn't warn about this skew, so users silently miss critical bugfixes.

## Reproduction

On Windows, vpcc 2.1.114 (installed via pipx some time ago), Claude Code 2.1.143:

```text
$ vpcc self-update
vpcc self-update  <- VoidChecksum/void-patcher-cc@main
  local  : 4c7dcf5bfc2ef315d154abf345362ea6826e6445
  remote : 4c7dcf5bfc2ef315d154abf345362ea6826e6445
OK already up to date

$ vpcc doctor
  vpcc ver   : 2.1.114
  patches    : 105
  upstream   : current
```

But the remote `pyproject.toml` says version 2.2.0:

```text
$ gh api repos/VoidChecksum/void-patcher-cc/contents/pyproject.toml \
    --jq '.content' | base64 -d | grep version
version         = "2.2.0"
```

And the remote `vpcc/__main__.py` has 1727 lines vs the local 1182 lines.

## Impact

The user is silently 5 releases behind. Several bug fixes that landed since 2.1.114 are not active:

- PR #17 fix for issue #6 (Windows PE `.bun` section padding -- uses `vsize or rsize` and trailer `rstrip(b"\x00")`). Without this, `patch_bun_sea_inplace` returns "Bun trailer invalid -- format change" on Windows builds where `SizeOfRawData > VirtualSize`.
- 2.1.132 cross-platform overhaul (commit e1873fc).
- 2.1.143-line scanner+autoheal perf overhaul (commit 52cf1fec, commit 4c7dcf5b).
- `_find_active_bundle_bounds` (only present on remote, narrows writable region inside the active entry bundle).

On Windows specifically, the missing PR #17 fix means `vpcc patch` falls into the Windows fallback path that skips non-required patches (see related issue I'm filing for that). The user reports "OK all patches verified" while five `required: false` patches are silently not applied to claude.exe.

## Proposed fix

`vpcc doctor` should detect the version skew and surface it:

```python
def _check_python_version_skew():
    try:
        import vpcc
        local_ver = getattr(vpcc, "__version__", None) or _read_local_pyproject_version()
    except Exception:
        return None
    remote_ver = _read_remote_pyproject_version()  # gh api repos/.../contents/pyproject.toml
    if local_ver and remote_ver and _semver_lt(local_ver, remote_ver):
        return (local_ver, remote_ver)
    return None

# in cmd_doctor:
skew = _check_python_version_skew()
if skew:
    local, remote = skew
    print(f"  {Y}vpcc package: local {local} < remote {remote} -- run `pipx upgrade vpcc`{X}")
```

Alternative: have `self-update` itself check the package version and print a hint when out of date, like "patches up to date, but vpcc package is at 2.1.114 vs remote 2.2.0 -- run `pipx upgrade vpcc` for the Python code".

## Why this matters more on Windows

Most Python-code bug fixes in 2.1.120 through 2.2.0 are Windows-specific (PE parsing, MSYS path handling, native-installer paths in the wrappers). Linux/macOS users get away with stale vpcc Python code because the patch JSON regenerations land first and the Python-side parsers worked fine for them already. Windows users carry the brunt of the skew.

## Workaround

Documenting for now: `pipx upgrade vpcc` after every `vpcc self-update` if you want both the JSON patches and the Python code current.

**Comment by ademczuk** [comment]
## Windows-user verification command

To detect if you're affected by the version skew silently, run from PowerShell:

```powershell
$localVer = (vpcc doctor 2>&1 | Select-String '^\s*vpcc ver').Line.Trim() -replace '.*:\s*',''
$remote = Invoke-RestMethod "https://raw.githubusercontent.com/VoidChecksum/void-patcher-cc/main/pyproject.toml"
$remoteVer = ($remote -split "`n" | Select-String 'version\s*=\s*"').Line -replace '.*"([^"]+)".*','$1'
"local vpcc: $localVer"
"remote   : $remoteVer"
if ($localVer -ne $remoteVer) {
  "SKEW DETECTED. Run: pipx upgrade vpcc"
} else {
  "in sync"
}
```

After running `pipx upgrade vpcc`, re-run the snippet. Both versions should match.

Caveat I hit during testing: `vpcc doctor` reports `vpcc ver: 2.1.120` even on a fresh install of vpcc 2.2.0 (pyproject says 2.2.0 but the version string inside `__init__.py` is stale). Either source-of-truth needs unifying, or `doctor` should read pyproject.toml at runtime. Adding that as a sub-issue here rather than filing separately: the version-skew detection only works if `vpcc doctor`'s reported version is reliable, and right now it isn't.

```text
$ pipx list --short | Select-String vpcc
vpcc 2.2.0

$ vpcc doctor | Select-String '^\s+vpcc ver'
  vpcc ver   : 2.1.120
```

Two reads, different answers. Pick a source-of-truth (recommend `pyproject.toml` via `importlib.metadata.version("vpcc")` at runtime).

**Comment by ademczuk** [comment]
Consolidated into the Windows 2.1.143 tracker at #23 (with proposed fix ordering and working regexes). Closing here to reduce noise on the tracker; reopen if the issue needs to be discussed separately.
